File: Java/Queue.java
import java.util.LinkedList;
public class Queue<T> implements Iterable<T> {
    private LinkedList<T> list = new LinkedList<T>();
    public Queue(){}
    public Queue(T elem){
        enqueue(elem);
    }
    public int size(){
        return list.size();
    }
    public boolean isEmpty(){
        return list.isEmpty();
    }
    public T peek(){
        if(isEmpty()){
            throw new RuntimeException("Queue Empty");
        }
        return list.peekFirst();
    }
    public void enqueue(T elem){
        list.addLast(elem);
    }
    public T dequeue(){
        if(isEmpty()){
            throw new RuntimeException("Queue Empty");
        }
        return list.removeFirst();
    }
    @Override public java.util.Iterator<T> iterator(){
        return list.iterator();
    }
}

File: DS_Implementations/Python/queue.py
from linked_list import *
class Queue:
    lst = LinkedList()
    def __init__(self,elem=None):
        if elem:
            self.offer(elem)
    def __len__(self):
        return len(self.lst)
    def isempty(self):
        return len(self.lst)==0
    def offer(self,elem):
        self.lst.addlast(elem)
    def poll(self):
        return self.lst.removefirst()
    def peek(self,index=0):
        return self.lst[index]
    def __getitem__(self,index):
        return self.peek(index)
    def __iter__(self):
        return self
    def __next__(self):
        return self.lst.__next__()
    def __repr__(self):
        return repr(self.lst)